x-pack/plugins/cases/server/custom_fields/factory.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CustomFieldTypes } from '../../common/types/domain';
+import type { ICasesCustomField, CasesCustomFieldsMap } from './types';
+import { getCasesTextCustomField } from './text';
+import { getCasesToggleCustomField } from './toggle';
+
+const mapping: Record<CustomFieldTypes, ICasesCustomField | null> = {
```

Comment:
Is it possible for the `getCasesTextCustomField` etc to return `null`?

Reply:
hmm.. Not 100% sure but in case we added new `CustomFieldTypes` and didn't add it's config here.

Reply:
Oh, I see what you mean. I think we would like for TS to let us know that we need to fill the mapping with our new type. Wdyt?

Reply:
yup, that works 👍  I will remove null.